azuredeploy.json

```diff
@@ -97,7 +97,7 @@
     "databaseSkuTier": "GeneralPurpose",
     "databaseSkuFamily": "Gen5",
     "databaseSkuName": "[concat('GP_', variables('databaseSkuFamily'), '_', parameters('databaseCores'))]",
-    "databaseConnectionString": "[concat('pgsql://', parameters('adminUserName'), '@', variables('databaseServerName'), ':', parameters('adminPassword'), '@', variables('databaseServerName'), '.postgres.database.azure.com:5432/', parameters('databaseName'))]",
+    "databaseConnectionString": "[concat('pgsql://', parameters('adminUserName'), '%40', variables('databaseServerName'), ':', parameters('adminPassword'), '@', variables('databaseServerName'), '.postgres.database.azure.com:5432/', parameters('databaseName'))]",
```

Comment:
there's a template function `uricomponent(string)` that will do a URL encoded.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-string#uricomponent

Reply:
Given that we only have a single well-known character to encode and that we have `%40` in one of the tests, I'd wager to say that not using the function would be more understandable/readable here.

Reply:
Done in 0811482.